pkg/migrations/sql/preinstall/006-set_up_ha.sql

```diff
@@ -0,0 +1,146 @@
+CREATE TABLE SCHEMA_CATALOG.ha_locks
+(
+    cluster_name TEXT PRIMARY KEY,
+    leader       TEXT,
+    lease_start  TIMESTAMPTZ,
+    lease_until  TIMESTAMPTZ
+);
+
+CREATE TABLE SCHEMA_CATALOG.ha_locks_log
+(
+    cluster_name TEXT        NOT NULL,
+    leader       TEXT        NOT NULL,
+    lease_start  TIMESTAMPTZ NOT NULL,
+    lease_until  TIMESTAMPTZ NOT NULL,
+    PRIMARY KEY (cluster_name, lease_start),
+    UNIQUE (cluster_name, leader, lease_start)
+);
+
+
+-- function that trigger to automatically keep the log calls
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.ha_locks_audit_fn()
+    RETURNS TRIGGER
+AS
+$func$
+BEGIN
+    INSERT INTO SCHEMA_CATALOG.ha_locks_log (cluster_name, leader, lease_start, lease_until)
+    VALUES (NEW.cluster_name, NEW.leader, NEW.lease_start, NEW.lease_until)
+           -- if lease is extended, but leader didn't change
+    ON CONFLICT (cluster_name, leader, lease_start) DO UPDATE SET lease_until=EXCLUDED.lease_until;
+    --no ON CONFLICT (cluster_name, leader) clause as there should never be conflicts
+    -- on the PK and if there are it's an error
+    RETURN NEW;
+END;
+$func$ LANGUAGE plpgsql VOLATILE;
+
+-- trigger to automatically keep the log
+CREATE TRIGGER ha_locks_audit
+    AFTER INSERT OR UPDATE
+    ON SCHEMA_CATALOG.ha_locks
+    FOR EACH ROW
+EXECUTE PROCEDURE SCHEMA_CATALOG.ha_locks_audit_fn();
+
+-- default values for lease
+INSERT INTO SCHEMA_CATALOG.default(key, value)
+VALUES ('ha_lease_timeout', '1m'),
+       ('ha_lease_refresh', '10s')
```

Comment:
Shouldn't these be values from CLI flags? This should really be configurable as timeouts are user dependent.

Reply:
This should not be a cli flag because if we have multiple connectors running, they cannot have different values. But, we should have DB functions to change these values

Reply:
But if you edit via DB functions, the moment the next fresh promscale instance is connected, it will update again?